### PR TITLE
OpenPaaS-Suite/esn-frontend-admin#42 - Bootstrap module

### PR DESCRIPTION
**Pull request by piavgh**
Partially resolves https://github.com/OpenPaaS-Suite/esn-frontend-admin/issues/42

### CODE REVIEW

**Review by itswillta — APPROVED** [review]
Please also have a PR to set up tests, CI, and ESLint for this repository also.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "esn-frontend-inbox-james",
+  "version": "1.0.0",
```
Comment (tuanlc) [inline package.json]:
```suggestion
  "version": "0.0.1",
```

File: package.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  "description": "Intermediary frontend module for Inbox and james",
```
Comment (tuanlc) [inline package.json]:
```suggestion
  "description": "Intermediary frontend module for Inbox and James",
```

File: package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  "devDependencies": {
+    "chai": "3.5.0",
+    "chai-as-promised": "5.3.0",
+    "chai-shallow-deep-equal": "1.4.0",
+    "eslint": "6.2.0",
+    "eslint-config-airbnb-base": "8.0.0",
+    "eslint-config-linagora-esn": "1.1.1",
```
Comment (tuanlc) [inline package.json]: use `@linagora/esnlint-config-linagora-esn@2.0.0` instead

File: src/all.less [hunk 4]
```diff
@@ -0,0 +1,2 @@
+@import "~esn-frontend-inbox/src/linagora.esn.unifiedinbox/app/inbox.less";
+@import "./app/app.less";
```
Comment (tuanlc) [inline src/all.less]: blank line after

File: src/app/app.config.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+angular.module('esn.inbox-james')
+  .config(registerI18N);
+
+function registerI18N($translateProvider) {
+  $translateProvider.translations('en', require('../i18n/en.json'));
+  $translateProvider.translations('fr', require('../i18n/fr.json'));
+  $translateProvider.translations('ru', require('../i18n/ru.json'));
+  $translateProvider.translations('vi', require('../i18n/vi.json'));
+  $translateProvider.translations('zh', require('../i18n/zh.json'));
+}
```
Comment (tuanlc) [inline src/app/app.config.js]: blank line after

File: src/app/app.module.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+var MODULE_NAME = 'esn.inbox-james';
```
Comment (tuanlc) [inline src/app/app.module.js]: s/var/const/g

File: src/app/common/inbox-james-restangular.service.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+var MODULE_NAME = 'esn.inbox-james';
+var MODULE_DIR_NAME = '/unifiedinbox.james';
```
Comment (tuanlc) [inline src/app/common/inbox-james-restangular.service.js]: s/var/const/g

File: src/app/deleted-messages/criteria/attachment/inbox-james-deleted-messages-criteria-attachment-form.pug [hunk 8]
```diff
@@ -0,0 +1,5 @@
+ng-form(name="form", ng-init="$ctrl.criterion.operator='equals'")
+  label.checkbox.checkbox-inline
+    input(type="checkbox", ng-model="$ctrl.criterion.value")
+    i.input-helper
+    | {{ 'Has attachment' | translate }}
```
Comment (tuanlc) [inline src/app/deleted-messages/criteria/attachment/inbox-james-deleted-messages-criteria-attachment-form.pug]: blank line after

File: src/app/deleted-messages/criteria/date/inbox-james-deleted-messages-criteria-date-form.pug [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    label.fg-label {{ 'the date' | translate }}
+    .fg-line
+      input.form-control.criterion-value(
+        type="text",
+        ng-model="$ctrl.criterion.value",
+        date-format="{{::$ctrl.dateFormat}}",
+        data-autoclose="1",
+        bs-datepicker,
+        required
+      )
```
Comment (tuanlc) [inline src/app/deleted-messages/criteria/date/inbox-james-deleted-messages-criteria-date-form.pug]: blank line after

File: src/app/deleted-messages/criteria/form/inbox-james-deleted-messages-criteria-form.pug [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+    delete-criterion="$ctrl.deleteCriterion($index)",
+    ng-if="!$ctrl.selectAll"
+  )
+
+  .no-criteria-message(ng-if="!$ctrl.criteria.length")
+    .message
+      .no-criteria-icon.mdi.mdi-format-list-bulleted
+      .message {{ 'No criterion, click + to add a new one' | translate }}
+
+fab(icon="plus", ng-if="!$ctrl.selectAll", ng-click="$ctrl.addCriterion()")
```
Comment (tuanlc) [inline src/app/deleted-messages/criteria/form/inbox-james-deleted-messages-criteria-form.pug]: blank line after

File: src/app/mail-repository/email/delete/inbox-james-mail-repository-email-delete-dialog.controller.js [hunk 11]
```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+angular.module('esn.inbox-james')
+  .controller('inboxJamesMailRepositoryEmailDeleteDialogController', inboxJamesMailRepositoryEmailDeleteDialogController);
+
+function inboxJamesMailRepositoryEmailDeleteDialogController(
+  $q,
+  _,
```
Comment (tuanlc) [inline src/app/mail-repository/email/delete/inbox-james-mail-repository-email-delete-dialog.controller.js]: Use `require` instead

File: src/app/mail-repository/email/inbox-james-mail-repository-email.model.js [hunk 12]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+angular.module('esn.inbox-james')
+  .factory('InboxJamesMailRepositoryEmail', InboxJamesMailRepositoryEmailFactory);
+
+function InboxJamesMailRepositoryEmailFactory(_) {
```
Comment (tuanlc) [inline src/app/mail-repository/email/inbox-james-mail-repository-email.model.js]: use `require` for `_`

File: src/app/mail-repository/inbox-james-mail-repository.controller.js [hunk 13]
```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+angular.module('esn.inbox-james')
+  .controller('inboxJamesMailRepositoryController', inboxJamesMailRepositoryController);
+
+function inboxJamesMailRepositoryController(
+  $scope,
+  _,
```
Comment (tuanlc) [inline src/app/mail-repository/inbox-james-mail-repository.controller.js]: same here

File: src/app/mail-repository/inbox-james-mail-repository.service.js [hunk 14]
```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+angular.module('esn.inbox-james')
+  .factory('inboxJamesMailRepository', inboxJamesMailRepository);
+
+function inboxJamesMailRepository(
+  _,
```
Comment (tuanlc) [inline src/app/mail-repository/inbox-james-mail-repository.service.js]: same here

File: src/app/mail-repository/selection/inbox-james-mail-repository-email-selection.service.js [hunk 15]
```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+angular.module('esn.inbox-james')
+  .factory('inboxJamesMailRepositoryEmailSelection', inboxJamesMailRepositoryEmailSelection);
+
+function inboxJamesMailRepositoryEmailSelection(_) {
```
Comment (tuanlc) [inline src/app/mail-repository/selection/inbox-james-mail-repository-email-selection.service.js]: same here for `_`

File: src/i18n/ru.json [hunk 16]
```diff
@@ -0,0 +1,111 @@
...
+  "the date": "la date",
+  "Delivery date": "Date de délivraison",
+  "Deletion date": "Date de suppression",
+  "Restored-Messages": "Messages restaurés",
+  "Report: Deleted messages restoration result": "Rapport: Bilan de la restauration des messages supprimés",
+  "Deleted messages restoration result": "Bilan de la restauration des messages supprimés",
+  "Number of successfully restored messages:": "Nombre de messages restaurés avec succès:",
+  "Number of failed restoring messages:": "Nombre de messages ayant échoué lors de la restauration:",
+  "You can find the restored messages inside mailbox": "Vous pouvez trouver vos messages restaurés dans votre boîte mail",
+  "In case of any restoration failure, please contact your domain administrator for more details.": "En cas d'échec de la restauration, veuillez contacter votre administrateur de domaine pour plus de détails."
```
Comment (tuanlc) [inline src/i18n/ru.json]: This file is not for `ru` but for `fr`

File: src/i18n/zh.json [hunk 17]
```diff
@@ -0,0 +1,111 @@
+{
+  "Admin Quarantine": "Zone de quarantaine administrative",
+  "Settings": "Paramètres",
+  "No rules, click": "Aucune règles, cliquez sur",
```
Comment (tuanlc) [inline src/i18n/zh.json]: Same for this file. Not for `zh` but for `fr`

**Comment by piavgh** [comment]
@tuanlc : all issues are resolved, please check again

**Review by tuanlc — APPROVED** [review]

**Pull request merged**